File: 04.SERVIDOR/WS_ComElect_RESTJAVA_GR05/src/main/java/ec/edu/yaranga/comerelect/dto/banquito/CreditoRequest.java
package ec.edu.yaranga.comerelect.dto.banquito;

import jakarta.validation.constraints.*;

import java.math.BigDecimal;

public record CreditoRequest(

        @NotBlank(message = "La cédula es obligatoria.")
        @Size(min = 10, max = 10, message = "La cédula debe tener 10 dígitos.")
        String cedula,

        @NotNull(message = "El precio del electrodoméstico es obligatorio.")
        @DecimalMin(value = "0.01", message = "El precio debe ser mayor a cero.")
        BigDecimal precioCredito,

        @NotNull(message = "El número de cuotas es obligatorio.")
        @Min(value = 3, message = "El número de cuotas debe ser al menos 3.")
        @Max(value = 24, message = "El número de cuotas debe ser de maximo 24.")
        Integer numeroCuotas

) {}

File: 04.CLI MOV/CLIMOV_ComerElect_RESTJAVA_GR05/app/src/main/java/ec/edu/monster/modelo/banquito/CreditoRequest.kt
package ec.edu.monster.modelo.banquito

import androidx.annotation.IntRange
import java.math.BigDecimal

data class CreditoRequest(
    val cedula: String,
    val precioCredito: BigDecimal,
    @IntRange(from = 3, to = 24)
    val numeroCuotas: Int
)
